src/phoenix/evals.py

```diff
@@ -0,0 +1,288 @@
+        raise NotImplementedError()
+
+    def predict_dataframe(
+        self, dataframe: pd.DataFrame, *, job_config: Optional[JobConfig] = None
+    ) -> List[Optional[ClassificationPrediction]]:
+        raise NotImplementedError()
+
+
+class OpenAIFunctionsClassifier:
+    def __init__(
+        self,
+        prompt_template: ClassificationPromptTemplate,
+        model: OpenAIModel,
+        function_name: str,
+        function_description: str,
+        class_argument_name: str,
+        class_argument_description: str,
+        *,
+        default_return_value: Optional[str] = None,
+    ) -> None:
+        raise NotImplementedError()
+
+    def predict(
+        self, record: Record, *, require_explanation: bool = False
+    ) -> ClassificationPrediction:
+        raise NotImplementedError()
+
+    def predict_batch(
+        self, record: Record, *, job_config: Optional[JobConfig] = None
+    ) -> List[Optional[ClassificationPrediction]]:
+        raise NotImplementedError()
+
+    def predict_dataframe(
+        self, dataframe: pd.DataFrame, *, job_config: Optional[JobConfig] = None
+    ) -> List[Optional[ClassificationPrediction]]:
+        raise NotImplementedError()
+
+
+# Support for fine-tuning.
+class OpenAIFineTuningJob:
+    def __init__(
+        self,
+        prompt_template: ClassificationPromptTemplate,
+        base_model_name: str,
+        *,
+        num_epochs: Optional[int] = None,
+    ) -> None:
+        raise NotImplementedError()
+
+    @property
+    def status(
+        self,
+    ) -> Literal["created", "pending", "running", "succeeded", "failed", "cancelled"]:
+        """Forwards the status from fine-tuning job endpoint."""
+        raise NotImplementedError()
+
+    @property
+    def model_name(self) -> Optional[str]:
+        """
+        Returns the name of the fine-tuned model if the job has succeeded,
+        otherwise, returns None.
+        """
+        raise NotImplementedError()
+
+    def create(self, data: List[Record]) -> JobID:
+        """Kicks off fine-tuning job."""
+        raise NotImplementedError()
+
+
+# Evaluators
+class Evaluator(Protocol):
+    def evaluate(self, records: List[Record]) -> List[Evaluation]:
+        ...
+
+    @property
+    def name(self) -> str:
+        ...
+
+
+class ClassificationEvaluator:
+    def __init__(self, classifier: LLMClassifier, name: str) -> None:
+        raise NotImplementedError()
+
+    def evaluate(self, records: List[Record]) -> List[Evaluation]:
+        """Delegates to the classifier."""
+        raise NotImplementedError()
+
+    @property
+    def name(self) -> str:
+        raise NotImplementedError()
+
+
+# not sure how to handle something like precision@k that requires an
+# classification evaluator before computing the score
+
+
+# evals
+class Evals:
+    def __init__(self, evaluators: List[Evaluator], job_config: Optional[JobConfig] = None) -> None:
+        raise NotImplementedError()
+
+    @classmethod
+    def from_names(
```

Comment:
not sure this path is needed - it foces two paths for construction

Reply:
Removed.